src/Cli/dotnet/commands/dotnet-run/VirtualProjectBuildingCommand.cs

```diff
@@ -117,6 +131,31 @@ public int Execute(string[] binaryLoggerArgs, ILogger consoleLogger)
         }
     }
 
+    /// <summary>
+    /// Needs to be called before the first call to <see cref="CreateProjectInstance(ProjectCollection)"/>.
+    /// </summary>
+    public VirtualProjectBuildingCommand PrepareProjectInstance()
+    {
+        Debug.Assert(_directives.IsDefault && _targetFilePath is null, $"{nameof(PrepareProjectInstance)} should not be called multiple times.");
+
+        var sourceFile = CreateSourceFile(EntryPointFileFullPath);
+        _directives = FindDirectives(sourceFile);
+
+        // If there were any `#:` directives, remove them from the file.
+        // (This is temporary until Roslyn is updated to ignore them.)
```

Comment:
Is there any reason not to remove them if roslyn would ignore them anyway? Just wondering why this wouldn't be permanent

Reply:
Because removing them changes line numbers. So for example location of errors reported by Roslyn is off.